sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/UpdateField.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.mongodb;
+
+import java.io.Serializable;
+
+public class UpdateField implements Serializable {
+
+  private String updateOperator;
+
+  private String sourceField;
+
+  private String destField;
+
+  public UpdateField(String updateOperator, String sourceField, String destField) {
+    this.updateOperator = updateOperator;
+    this.sourceField = sourceField;
+    this.destField = destField;
+  }
+
+  /** for updating field by field. */
+  public static UpdateField of(String updateOperator, String sourceField, String destField) {
+    return new UpdateField(updateOperator, sourceField, destField);
+  }
+
+  /** for updating with entire input document. */
+  public static UpdateField of(String updateOperator, String destField) {
+    return new UpdateField(updateOperator, null, destField);
```

Comment:
is this used whenever we create a new field? Or whenever the source and dest fields have the same name?

Reply:
I wonder if it makes sense to add an option for whenever source and destination field have the same name : )

Reply:
This is used when we need to update full source document into destination document. (Not updating specific fields). Whenever we have to update specific fields we need to specify which field in the source has to be updated to which field in dest document. Do you think there can be a better way?

Reply:
@pabloem Have changed it to auto value class which has two methods - fieldUpdate and fullUpdate. Example below:
MongoDbIO.write()
.withUri("mongodb://localhost:" + port)
.withDatabase(DATABASE)
.withCollection(collectionName)
.withUpdateConfiguration(
UpdateConfiguration.create()
.withUpdateKey("id")
.withUpdateFields(
UpdateField.create().fieldUpdate("$set", "scientist", "scientist"),
UpdateField.create().fieldUpdate("$set", "country", "country"))));